### PR TITLE
Integrate new motor driver and gyro

### PR DESCRIPTION
**Pull request by Peque**
Very basic integration, but seems to work fine. I thought I would ask for review now, before it gets harder... :blush:

### CODE REVIEW

File: src/setup.c [hunk 1]
```diff
@@ -271,61 +271,6 @@ static void setup_adc1(void)
 	adc_calibrate(ADC1);
 }
 
-/**
```
Comment (cua-cua) [inline src/setup.c]: I would keep this function to read battery level, removing the old gyroscope configuration. So I created a new issue #210 to update this function with new hardware changes.

File: docs/source/configuration.rst [hunk 2]
```diff
@@ -73,13 +83,13 @@ PB8   GPIO       Output  Right side emmiter switch
...
+PB12  NSS2       Output  Gyroscope NCS (SPI-CS)
+PB13  SCK2       Output  Gyroscope SCL (SPI-SCLK)
+PB14  MISO2      Input   Gyroscope AD0 (SPI-SDO)
+PB15  MOSI2      Output  Gyroscope SDA (SPI-SDI)
+PC13  N/A        N/A     *Not connected*
+PC14  N/A        N/A     *Not connected*
+PC15  N/A        N/A     *Not connected*
 ====  =========  ======  ==============================
 
 Motor driver
```
Comment (cua-cua) [inline docs/source/configuration.rst]: Good idea to include not only the bluepill pinout :)

File: src/setup.c [hunk 3]
```diff
@@ -120,6 +123,54 @@ static void setup_usart(void)
 	usart_enable(USART3);
 }
 
+/**
+ * @brief Setup SPI for gyroscope communication.
+ *
+ * SPI operational features of the gyroscope:
+ *
+ * - Data is delivered MSB first and LSB last.
```
Comment (cua-cua) [inline src/setup.c]: I would add this first paragraph on documentation section

Comment (Peque) [reply src/setup.c]: Yeah, good idea. Moved that to the documentation. :blush:

File: docs/source/configuration.rst [hunk 4]
```diff
@@ -37,48 +37,111 @@ to the `IE2-512 encoder`_ channels A and B). They are both configured to count
...
+- Read/write operations are completed in 16+ cycles (2+ bytes).
+- First byte contains the SPI address, and the following the SPI data.
+- First bit of the first byte contains the Read (1) / Write (0) bit.
+- The following 7 bits contain the Register Address.
+- In cases of multiple-byte read/writes, data is two or more bytes.
+
 .. index:: pinout
 
 Pinout
 ======
```
Comment (cua-cua) [inline docs/source/configuration.rst]: Pinout reviewed and coherent with protoboard and schematic

**Comment by Peque** [comment]
@cua-cua Sure! I will wait for you to merge that PR into master first. Then I will rebase this one. :+1:

**Comment by cua-cua** [comment]
Gyro cofiguration removed and merged. Master, ready to rebase.

**Pull request merged**